drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,452 @@
+import functools
+import itertools
+import json
+import logging
+import multiprocessing
+import pathlib
+import shutil
+import subprocess
+import zipfile
+from typing import Any, BinaryIO, Dict, Iterator, List, Optional, Tuple, Union
+
+import capa
+import capa.capabilities
+import capa.capabilities.common
+import capa.engine
+import capa.features
+import capa.features.address
+import capa.features.common
+import capa.features.extractors
+import capa.features.extractors.base_extractor
+import capa.features.extractors.drakvuf
+import capa.features.extractors.drakvuf.extractor
+import capa.helpers
+import capa.loader
+import capa.main
+import capa.render
+import capa.render.json
+import capa.render.result_document
+import capa.rules
+import msgspec.json
+
+# rules related configuration
+capa_rules_dir = pathlib.Path("./capa-rules")
+capa_default_rules_repository_url = "https://github.com/mandiant/capa-rules/"
+
+# analysis configuration
+analyze_malware_pids_only = False
+perform_static_analysis = False
+perform_dynamic_analysis = True
+
+logger = logging.getLogger(__name__)
+
+
+def check_rules_exist(rules_dir: pathlib.Path) -> bool:
+    # this method checks whether there is a non-empty capa rules folder
+    try:
+        if any(rules_dir.iterdir()):
+            # return true if there exists a non-empty rules folder
+            return True
+        else:
+            # if the folder is empty, remove it and return false
+            shutil.rmtree(rules_dir)
+
+    except FileNotFoundError:
+        # if the rules folder does not exist, return false
+        pass
+
+    except NotADirectoryError:
+        # if there's a file with the same name as the rules' directory, delete it and return false
+        rules_dir.unlink()
+
+    # return false if no suitable capa rules folder was found
+    return False
+
+
+def clone_rules_reporsitory(
+    rules_dir: pathlib.Path, rules_git_repo_url=capa_default_rules_repository_url
+):
+    logger.info("Cloning default capa rules repository into %s", rules_dir.absolute)
+
+    # clone the official capa rules repository
+    subprocess.run(
+        f"git clone {rules_git_repo_url} {pathlib.Path}",
+        shell=True,
+        stderr=subprocess.STDOUT,
+    )
```

Comment:
> Also you probably meant {rules_dir} instead of {pathlib.Path}.

Oh yeah true, sorry I didn't test this part 😅

Thanks for the tips on using arrays with subprocess! I'll get rid of this routine for now.